### PR TITLE
Makes HE pipes realistically radiate away heat

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

1. Heat exchange now also radiate heat proportional to their blackbody radiation. This makes space cooling loops WAY stronger at >6000 kelvins, to the point that it's now feasible to cool down fusion cans and such. Of course, this also makes HE pipes being used as weapons stronger.
2. Makes HE pipes not special case lava tiles when lava tiles already have a special case temperature get that HE pipes use anyway.
3. Makes the ideal gas equation more precise (before 8.31, now 8.31446261815324, its exact value in SI, though it should be noted that some of this precision is lost to floating point math).

I thought about making space cooling loops *only* work based on radiation, like IRL, but it turns out that would make them *extremely* weak at even lowish temperatures (e.g. 3000x as weak at fire temperatures), so nah.

## Why It's Good For The Game

Fusion cans shouldn't be eternal heat sinks impossible to fix.

## Changelog
:cl:
balance: Buffed HE pipes by making them realistically radiate away heat.
/:cl:

### CODE REVIEW

File: code/__DEFINES/atmospherics.dm [hunk 1]
```diff
@@ -149,9 +150,9 @@
 
 //OPEN TURF ATMOS
 #define OPENTURF_DEFAULT_ATMOS		"o2=22;n2=82;TEMP=293.15" //the default air mix that open turfs spawn
-#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
+#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
```
Comment (Putnam3145) [inline code/__DEFINES/atmospherics.dm]: wow thanks

Comment (Ghommie) [reply code/__DEFINES/atmospherics.dm]:
```suggestion
#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15°C telecommunications. also used for xenobiology slime killrooms
```
Sucks, locales and lack of UTF-8 support whatever.

File: code/__DEFINES/atmospherics.dm [hunk 2]
```diff
@@ -149,9 +150,9 @@
 
 //OPEN TURF ATMOS
 #define OPENTURF_DEFAULT_ATMOS		"o2=22;n2=82;TEMP=293.15" //the default air mix that open turfs spawn
-#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
+#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
 #define AIRLESS_ATMOS				"TEMP=2.7" //space
-#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15�C snow and ice turfs
+#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15�C snow and ice turfs
```
Comment (Ghommie) [inline code/__DEFINES/atmospherics.dm]:
```suggestion
#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15°C snow and ice turfs
```

File: code/modules/atmospherics/machinery/datum_pipeline.dm [hunk 3]
```diff
@@ -154,6 +154,11 @@
 	var/partial_heat_capacity = total_heat_capacity*(share_volume/air.volume)
 	var/target_temperature
 	var/target_heat_capacity
+	// first calculate heat from radiation. there's an implied "* 1 tick" here.
+	// 0.05 magic multiplicand is, first, 0.1 deciseconds; second, half of the radiation's going right back into the gas.
+	var/share_constant = STEFANBOLTZMANN*(share_volume**(2/3))*0.05
+	// Minimizing temp to 4 billion is mostly to prevent -infinity temperatures.
+	var/heat = share_constant*(min(air.temperature,4000000000)**4)
```
Comment (Ghommie) [inline code/modules/atmospherics/machinery/datum_pipeline.dm:161]: I think these should be static at the least, but the PR has been open for a while and I entrust your knowledge of heat radiation or osmosis is not gonna break things.

**Review by Ghommie — APPROVED** [review]

**Comment by ghost** [comment]
YOU FUCKED IT DUMBASS

**Pull request merged**